### PR TITLE
test: improve handling of pending pods crashloop detection test

### PR DESCRIPTION
**Pull request by tsmetana**
The crashlooping test is also detecting pods that were pending for the entire run of the test and fails if such a pod is detected. I think the logic behind it is wrong: it creates the list of pending pods every 5 seconds so at the end of the test it is incorrect to claim the pod was pending entire time. I'm adding a time stamp comparison that should make this conclusion valid.

Also -- I don't think it's completely corect to refer to the pods by "namespace/pod_name" in this case: some pods might live for really short time and even get re-created between the polls. Hence the change to use UID instead.

### CODE REVIEW

**Comment by tsmetana** [comment]
@deads2k Could you please take a look at the changes? Do they make sense? I really think the test is not doing exactly what it says...

File: test/extended/operators/cluster.go [hunk 1]
```diff
@@ -136,10 +137,14 @@ func crashloopingContainerCheck(podFilters ...podFilter) {
 			e2e.Logf("Pod status %s/%s ignored for being pending", pod.Namespace, pod.Name)
 			continue
 		}
+		if pod.Status.StartTime.After(testStartTime) {
```
Comment (deads2k) [inline test/extended/operators/cluster.go:140]: This check should not be present.  A pod that was started before this test began (which is random in time to start) should definitely have time to complete its lifecycle if it is pending.

Comment (deads2k) [inline test/extended/operators/cluster.go:140]: Is this comment accurate?

```
		// this point lastPending has a list of pods that were pending a few seconds ago, but those pods could have been created
		// a few seconds before that.  What we really want is to know which pods are pending for longer than a predetermined threshold
		// of time.  This test implies that the intent is "find pods which have been pending more than four minutes.
```

Comment (tsmetana) [reply test/extended/operators/cluster.go:140]: Yes, it is.

Comment (tsmetana) [reply test/extended/operators/cluster.go:140]: OK... I'll add the comment.

**Comment by deads2k** [comment]
/lgtm

but I would like to get a better test for pending pods into the monitor I think

/cherry-pick release-4.6

**Pull request merged**